src/main/java/org/springframework/data/r2dbc/dialect/Dialect.java

```diff
@@ -0,0 +1,31 @@
+package org.springframework.data.r2dbc.dialect;
+
+/**
+ * Represents a dialect that is implemented by a particular database.
+ *
+ * @author Mark Paluch
+ */
+public interface Dialect {
+
+	/**
+	 * Returns the {@link BindMarkersFactory} used by this dialect.
+	 *
+	 * @return the {@link BindMarkersFactory} used by this dialect.
+	 */
+	BindMarkersFactory getBindMarkersFactory();
+
+	/**
+	 * Returns the statement to include for returning generated keys. The returned query is directly appended to
+	 * {@code INSERT} statements.
+	 *
+	 * @return the statement to include for returning generated keys.
+	 */
+	String returnGeneratedKeys();
```

Comment:
We will remove `returnGeneratedKeys` with R2DBC 1.0M7 again. For now, we're using it because right now returning generated keys requires some additional SQL. See https://github.com/r2dbc/r2dbc-spi/issues/25.